packages/commonwealth/client/scripts/views/components/linked_addresses.tsx

```diff
@@ -69,27 +101,53 @@ export const LinkedAddresses = (props: LinkedAddressesProps) => {
 
   const { profile, addresses, refreshProfiles } = props;
 
+  const groupedAddresses = useMemo(() => {
+    return addresses.reduce((acc: Record<string, AddressInfo[]>, addr) => {
+      if (!acc[addr.address]) acc[addr.address] = [];
+      acc[addr.address].push(addr);
+      return acc;
+    }, {});
+  }, [addresses]);
+
+  const rowData = Object.entries(groupedAddresses).map(
+    ([address, communities]) => ({
+      address: <Address address={address} />,
+      communities: (
+        <div>
+          {communities.map((addr, index) => {
+            return (
+              <AddressDetails
+                key={index}
+                profile={profile}
+                addressInfo={addr}
+                toggleRemoveModal={(
+                  val: boolean,
+                  selectedAddress: AddressInfo,
+                ) => {
+                  setIsRemoveModalOpen(val);
+                  setCurrentAddress(selectedAddress);
+                }}
+              />
+            );
+          })}
+        </div>
+      ),
+    }),
+  );
+
+  const TableComponent = useMemo(() => {
+    return <CWTable columnInfo={columnInfo} rowData={rowData} />;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [addresses]);
+
   return (
-    <div className="LinkedAddresses">
-      {addresses?.map((addr, i) => {
-        return (
-          <Address
-            key={i}
-            profile={profile}
-            addressInfo={addr}
-            toggleRemoveModal={(val: boolean, address: AddressInfo) => {
-              setIsRemoveModalOpen(val);
-              setCurrentAddress(address);
-            }}
-          />
-        );
-      })}
+    <div>
+      {TableComponent}
```

Comment:
let's just use CWTable here

Reply:
useMemo is used to memoize the CWTable component. This optimisation prevents unnecessary recalculations of the table structure and improves  performance.

```
  const TableComponent = useMemo(() => {
    return <CWTable columnInfo={columnInfo} rowData={rowData} />;
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [addresses]);
```